### ISSUE TITLE
Pysp throw keyError exception with block varible

### ISSUE THREAD

**Issue by wanqiuchansheng** [Pyomo/pyomo#718]
i have a complex model with large varibles defined in block. and i want to use pysp ph alghortim to solve it. and a simple model is flowing.
Traceback (most recent call last):
  File "/home/administrator/project/python/gym-rl/pyomotools/test/pysp/simple_quadratic/run.py", line 12, in <module>
    ef_sol = stsolver.solve_ph("gurobi", phopts=phopts, default_rho=1)
  File "/home/administrator/project/python/gym-rl/pyomotools/rapper.py", line 227, in solve_ph
    retval = ph.solve()
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/ph.py", line 4081, in solve
    self._report_only_nonconverged_variables)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/ph.py", line 4874, in pprint
    stage)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/ph.py", line 4858, in _print_stage_var
    num_outputs_this_variable=num_outputs_this_variable)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/ph.py", line 4680, in _print_node_var
    variable_indices = tree_node._variable_indices[variable_name]
KeyError: 'v1'

https://github.com/Pyomo/pyomo/blob/67696dfa2797c3db1195f84ee9d2439ee6c6a964/pyomo/pysp/ph.py#L4680

![image](https://user-images.githubusercontent.com/9635433/47158006-3a465400-d31d-11e8-888a-6fb5693c7f49.png)
ReferenceModel_new.py
```python
from pyomo.environ import *

scenario_data = {}
scenario_data['Scenario1'] = (2.1, 23.5)
scenario_data['Scenario2'] = (0.2, 4.2)
scenario_data['Scenario3'] = (0.6, 0.1)
scenario_data['Scenario4'] = (1.1, -4.4)

def pysp_scenario_tree_model_callback():
    from pyomo.pysp.scenariotree.tree_structure_model \
        import CreateConcreteTwoStageScenarioTreeModel

    st_model = CreateConcreteTwoStageScenarioTreeModel(len(scenario_data))

    first_stage = st_model.Stages.first()
    second_stage = st_model.Stages.last()

    # First Stage
    st_model.StageCost[first_stage] = 'FirstStageCost'
    st_model.StageVariables[first_stage].add('v1')
    st_model.StageVariables[first_stage].add('xd')
    st_model.StageDerivedVariables[first_stage].add('xde')

    # Second Stage
    st_model.StageCost[second_stage] = 'SecondStageCost'
    st_model.StageVariables[second_stage].add('y')
    st_model.StageVariables[second_stage].add('v2')
    return st_model


# Creates an instance for each scenario
def pysp_instance_creation_callback(scenario_name, node_names):
    a, b = scenario_data[scenario_name]

    model = ConcreteModel()
    model.v1 = Block()
    model.v1.x = Var()
    model.y = Var()
    model.FirstStageCost = Expression(expr=model.v1.x ** 2)
    model.SecondStageCost = Expression(expr=a * (model.y - b) ** 2)
    model.obj = Objective(expr=model.FirstStageCost + model.SecondStageCost)

    model.xd = Var()
    model.xde = Expression(expr=model.v1.x)
    # model.yd = Var()
    # model.yde = Expression(expr=model.y - b)

    model.v2 = Block()
    v2 = model.v2
    v2.yd_test = Var()
    v2.yde = Expression(expr=model.y - b)

    model.c = ConstraintList()
    model.c.add(model.y == model.v1.x)
    model.c.add(model.xd == model.v1.x)
    model.c.add(v2.yd_test == model.y - b)
    return model
```
run.py
``` python
phopts = {'--max-iterations': '3', '--verbose': None}
# phopts = {'--max-iterations': '3'}
stsolver = rapper.StochSolver("ReferenceModel_new.py")
ef_sol = stsolver.solve_ph("gurobi", phopts=phopts, default_rho=1)
print(ef_sol)

```

**Comment by blnicho** [comment]
This seems like additional information for #714. Please continue the discussion there instead of opening new issues for the same problem.

**Comment by DLWoodruff** [comment]
Thanks for the report. It will be a few weeks before we can address this.
  Dave

**Comment by wanqiuchansheng** [comment]
> This seems like additional information for #714. Please continue the discussion there instead of opening new issues for the same problem.

hi @blnicho, thansk for your reply.  #714  is not  the same, i have resolve #714 problem, it is due to rapper's bug.  maybe you can close #714 and remain this.

**Comment by blnicho** [comment]
@wanqiuchansheng my mistake.

**Comment by blnicho** [comment]
@DLWoodruff was this resolved in #727?

**Comment by ghackebeil** [comment]
No. I’m still working on a fix.
